pkgs/applications/science/electronics/horizon-eda/default.nix

```diff
@@ -45,14 +48,36 @@ stdenv.mkDerivation rec {
     libuuid
     opencascade-occt
     podofo
-    python3
     sqlite
+  ] ++ lib.optionals (pythonSupport) [
+    mesa
+    mesa.osmesa
+    python3Packages.python
+  ];
+
+  propagatedBuildInputs = lib.optionals (pythonSupport) [
+    python3Packages.pycairo
   ];
 
   nativeBuildInputs = [
     boost.dev
     pkg-config
     wrapGAppsHook
+  ] ++ lib.optionals (!pythonSupport) [
+    python3
+  ];
+
+  postBuild = lib.optionalString (pythonSupport) ''
+    make -j$NIX_BUILD_CORES pymodule
+  '';
+
+  postInstall = lib.optionalString (pythonSupport) ''
+    mkdir -p $out/${python3Packages.python.sitePackages}
+    cp build/horizon.so $out/${python3Packages.python.sitePackages}
+  '';
+
+  pythonImportsCheck = lib.optionals (pythonSupport) [
+    "horizon"
   ];
```

Comment:
For this to run you likely have to include `python3Packages.pythonImportsCheckHook` as a native build input. Verify it works by changing it to something erroneous. ;)

Reply:
Thank you for the hint! My patch was very try-and-error-driven, tbh :D

Reply:
I tried that, but import wasn't possible. Maybe my `installPhase` is too hacky?